// danarrr-uidesign/dist/component/Skeleton/SkeletonBox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _prefix = require('./../../_utils/prefix');

var _prefix2 = _interopRequireDefault(_prefix);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SkeletonBox = function (_Component) {
    _inherits(SkeletonBox, _Component);

    function SkeletonBox(props) {
        _classCallCheck(this, SkeletonBox);

        return _possibleConstructorReturn(this, (SkeletonBox.__proto__ || Object.getPrototypeOf(SkeletonBox)).call(this, props));
    }

    _createClass(SkeletonBox, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                style = _props.style,
                prefixCls = _props.prefixCls,
                className = _props.className,
                justify = _props.justify,
                align = _props.align,
                direction = _props.direction,
                children = _props.children;

            var selectClassName = (0, _classnames2.default)(prefixCls, className);

            return _react2.default.createElement(
                'div',
                { style: _extends({}, style, { alignItems: align, justifyContent: justify, flexDirection: direction }), className: selectClassName },
                children
            );
        }
    }]);

    return SkeletonBox;
}(_react.Component);

exports.default = SkeletonBox;


SkeletonBox.defaultProps = {
    prefixCls: _prefix2.default.cls + '-skeleton-box',
    className: '',
    style: {},
    justify: 'flex-start',
    align: 'flex-start',
    direction: "row"
};

SkeletonBox.propTypes = {
    prefixCls: _propTypes2.default.string,
    className: _propTypes2.default.string,
    style: _propTypes2.default.object,
    justify: _propTypes2.default.string,
    align: _propTypes2.default.string,
    direction: _propTypes2.default.string
};
